applications/asset_tracker/src/main.c

```diff
@@ -372,6 +379,60 @@ static void flip_send(struct k_work *work)
 	}
 }
 
+/**@brief Callback handler for LTE RSRP data. */
+static void modem_rsrp_handler(char *rsrp_value, size_t len)
+{
+	static u32_t timestamp_prev;
+
+	if (!atomic_get(&send_data_enable)) {
+		return;
+	}
+
+	if (k_uptime_get_32() - timestamp_prev <
+	K_SECONDS(CONFIG_HOLD_TIME_RSRP)) {
+		return;
+	}
+
+	signal_strength_cloud_data.data.ptr = rsrp_value;
+	signal_strength_cloud_data.data.len = len;
+	signal_strength_cloud_data.tag += 1;
+
+	if (signal_strength_cloud_data.tag == 0) {
+		signal_strength_cloud_data.tag = 0x1;
+	}
+
+	sensor_data_send(&signal_strength_cloud_data);
+	timestamp_prev = k_uptime_get_32();
+}
+
+/**@brief Poll device info and send data to the cloud. */
+static void device_status_send(struct k_work *work)
+{
+	int len;
+	char data_buffer[DEVICE_INFO_STRING_SIZE] = {0};
+
+	if (!atomic_get(&send_data_enable)) {
+		return;
+	}
+
+	len = device_info_get_json_string(data_buffer);
+	if (len < 0) {
+		return;
+	}
+
+	device_cloud_data.data.ptr = data_buffer;
+	device_cloud_data.data.len = len;
+	device_cloud_data.tag += 1;
+
+	if (device_cloud_data.tag == 0) {
+		device_cloud_data.tag = 0x1;
+	}
+
+	sensor_data_send(&device_cloud_data);
+
+	modem_info_rsrp_sub_init(modem_rsrp_handler);
```

Comment:
I agree with this statement, however, the current implementation is a workaround. 

A weakness with the modem informer is that both RSRP subscription and normal modem data acquisition share the same socket. I believe this would result in a race condition, so in order to mitigate this is subscription first initiated after the initial data transmission. (However, in the future if the cloud requests data from the modem informer would the condition be a problem again).

I suggest one of the following:

1. Keep the current implementation until a common at_cmd - module is implemented, which will solve this issue.
2. Go for your suggestion, initialize RSRP subscription upon modem_info init. (But this could lead to strange behavior without a proper fix.)
3. Develop a solid solution for said issue, however this will also be solved over time based on the work on point 1.